lib/linked_list.rb

```diff
@@ -0,0 +1,88 @@
+# - Build a linked list node that contains the word and its definition
+# (and anything else you need) using a Struct (not a class). [ok]
+# - Build a basic linked list class that allows you to:
+#   1. Create the list [ok]
+#   2. Read nodes at a particular index (add a printout which tracks the
+#   crawler's progress or at least returns the number of steps). In the
+#   comments, state the Big-O time of this operation.
+#   3. Insert nodes at a particular index or, separately, at the end of
+#   the list (so store a reference to the last node). In the comments,
+#   state the Big-O time of this operation.
+# - Build a method #reverse on your LinkedList class that will entirely
+# flip the order of this list, so the "head" and the "tail" are swapped
+# and all the links point in opposite directions.
+# - In a comment above the method, explain how quickly it runs in Big O
+# notation. Is it "in place" (you reused the existing nodes), or did you
+# have to create new nodes to replace the old ones?
+
+Node = Struct.new("Node", :first, :rest)
+
+class LinkedList
+  attr_accessor :child
+
+  def initialize(child = Node.new)
+    @child = child
```

Comment:
It is conventional to name the first node in a linked list as `head` and last node as `tail`.